core/src/test/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotServiceSpec.scala

```diff
@@ -87,14 +80,33 @@ class SnapshotServiceSpec extends AnyWordSpecLike with Matchers with MockitoSuga
     mockWorkspaceManagerDAO
   }
 
+  // create a mockito-powered DataRepoDAO that always returns a stubbed snapshot
+  private def defaultDataRepoDao(): DataRepoDAO = {
+    val mockDataRepoDAO = mock[DataRepoDAO](RETURNS_SMART_NULLS)
+    when(
+      mockDataRepoDAO.getSnapshot(
+        any[UUID],
+        any[OAuth2BearerToken]
+      )
+    )
+      .thenReturn(
+        new SnapshotModel()
+          .id(java.util.UUID.randomUUID())
+          .name("snapshot")
+          .description("snapshot description")
+          .source(
+            java.util.List
+              .of(new SnapshotSourceModel().dataset(new DatasetSummaryModel().cloudPlatform(SnapshotCloudPlatform.GCP)))
```

Comment:
Now that we actively check the snapshot -> dataset -> cloudPlatform, this object needs to be adequately hydrated for all the existing tests to continue passing.